File: app/java/src/main/java/com/example/production/domain/model/process/WorkOrder.java
package com.example.production.domain.model.process;

import com.example.production.domain.model.item.Item;
import com.example.production.domain.model.location.Location;
import com.example.production.domain.model.plan.Order;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class WorkOrder {
    private Integer id;
    private String workOrderNumber;
    private String orderNumber;
    private LocalDate workOrderDate;
    private String itemCode;
    private BigDecimal orderQuantity;
    private String locationCode;
    private LocalDate plannedStartDate;
    private LocalDate plannedEndDate;
    private LocalDate actualStartDate;
    private LocalDate actualEndDate;
    @Builder.Default
    private BigDecimal completedQuantity = BigDecimal.ZERO;
    @Builder.Default
    private BigDecimal totalGoodQuantity = BigDecimal.ZERO;
    @Builder.Default
    private BigDecimal totalDefectQuantity = BigDecimal.ZERO;
    @Builder.Default
    private WorkOrderStatus status = WorkOrderStatus.NOT_STARTED;
    @Builder.Default
    private Boolean completedFlag = false;
    private String remarks;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;

    // リレーション
    private Order order;
    private Item item;
    private Location location;
    private List<WorkOrderDetail> details;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Process/WorkOrder.cs
namespace ProductionManagement.Domain.Models.Process;

/// <summary>
/// 作業指示データ
/// </summary>
public class WorkOrder
{
    private string _statusValue = "未着手";

    public int Id { get; set; }
    public required string WorkOrderNumber { get; set; }
    public required string OrderNumber { get; set; }
    public required DateOnly WorkOrderDate { get; set; }
    public required string ItemCode { get; set; }
    public required decimal OrderQuantity { get; set; }
    public required string LocationCode { get; set; }
    public required DateOnly PlannedStartDate { get; set; }
    public required DateOnly PlannedEndDate { get; set; }
    public DateOnly? ActualStartDate { get; set; }
    public DateOnly? ActualEndDate { get; set; }
    public decimal CompletedQuantity { get; set; }
    public decimal TotalGoodQuantity { get; set; }
    public decimal TotalDefectQuantity { get; set; }

    public string StatusValue
    {
        get => _statusValue;
        set => _statusValue = value;
    }

    public WorkOrderStatus Status
    {
        get => WorkOrderStatusExtensions.FromDisplayName(_statusValue);
        set => _statusValue = value.GetDisplayName();
    }

    public bool CompletedFlag { get; set; }
    public string? Remarks { get; set; }
    public DateTime? CreatedAt { get; set; }
    public string? CreatedBy { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string? UpdatedBy { get; set; }

    public IReadOnlyList<WorkOrderDetail> Details { get; set; } = [];
}
